src/sonic-build-hooks/hooks/git

```diff
@@ -45,17 +45,15 @@ get_clone_path(){
 
 main(){
     parse_config "$@"
-
-    # execute git.
-    $REAL_COMMAND "$@"
-    result=$?
+    get_clone_path "$@"
 
     # if sub command is not "clone", exit
     if [[ $MODE_CLONE != 1 ]];then
-        exit $result
+        $REAL_COMMAND "$@"
+        exit
     fi
+    run_with_retry $REAL_COMMAND "$@" || exit $?
```

Comment:
Could you please make it configurable? It is only used by Azp by default, for the other users, need to change the command lines option (rules/config) to enable it.